secret.go

```diff
@@ -30,10 +31,22 @@ func getSecret(ctx context.Context, destNamespace string, secret secretInfo) (*v
 	// check if app's destination namespace is allowed on given secret resource
 	for _, v := range strings.Split(sec.Annotations[secretAllowedNamespacesAnnotation], ",") {
 		if strings.TrimSpace(v) == destNamespace {
-			return sec, nil
+			return isSecretEncrypted(sec)
 		}
 	}
 
 	return nil, fmt.Errorf(`secret "%s/%s" cannot be used in namespace "%s", the destination namespace must be listed in the '%s' annotation`,
 		secret.namespace, secret.name, destNamespace, secretAllowedNamespacesAnnotation)
 }
+
+// isSecretEncrypted will go through all keys of the secret passed
+// and error out if at least one of them is encrypted
+func isSecretEncrypted (sec *v1.Secret) (*v1.Secret, error) {
```

Comment:
`isSecretEncrypted` - suggests return type is a boolean, if its returning secret + error - I would suggest `verifySecretEncrypted`